jbake-core/src/test/java/org/jbake/util/HtmlUtilTest.java

```diff
@@ -40,34 +40,50 @@ public void shouldNotAddBodyHTMLElement(){
 		
 	}
 
+    @Test
+    public void shouldNotAddSiteHost(){
+        Map<String, Object> fileContent = new HashMap<String, Object>();
+        fileContent.put(Attributes.ROOTPATH, "../../../");
+        fileContent.put(Attributes.URI, "blog/2017/05/first_post.html");
+        fileContent.put(Attributes.BODY, "<div> Test <img src='./first.jpg' /></div>");
+        config.setProperty(ConfigUtil.Keys.IMG_PATH_PREPEND_HOST,false);
+
+        HtmlUtil.fixImageSourceUrls(fileContent, config);
+
+        String body = fileContent.get(Attributes.BODY).toString();
+
+        assertThat(body).contains("src=\"../../../blog/2017/05/first.jpg\"");
```

Comment:
This will only work for the first_post.html file. if it is an post shown with content on the index page the url would be wrong openening the file directly from a browser. on the index page it should be `blog/2017/05/first.jpg`.